src/string_bytes.cc

```diff
@@ -857,7 +856,15 @@ Local<Value> StringBytes::Encode(Isolate* isolate,
                                  const uint16_t* buf,
                                  size_t buflen) {
   Local<String> val;
-
+  std::vector<uint16_t> dst(buflen);
```

Comment:
I'd declare an empty std::vector here and .resize() it inside the if block.  That avoids the memory allocation on little-endian architectures.